rewrite-java/src/main/java/org/openrewrite/java/tree/J.java

```diff
@@ -1027,14 +1027,27 @@ public Case withPattern(@Nullable Expression pattern) {
             return withExpressions(ListUtils.mapFirst(getExpressions(), first -> pattern));
         }
 
-        JContainer<Expression> expressions;
-
         public List<Expression> getExpressions() {
-            return expressions.getElements();
+            return caseLabels != null ? caseLabels.getElements().stream().filter(Expression.class::isInstance).map(Expression.class::cast).collect(toList()) : emptyList();
         }
 
         public Case withExpressions(List<Expression> expressions) {
-            return getPadding().withExpressions(requireNonNull(JContainer.withElementsNullable(this.expressions, expressions)));
+            if (caseLabels != null && caseLabels.getElements().stream().allMatch(Expression.class::isInstance)) {
+                return getPadding().withCaseLabels(requireNonNull(JContainer.withElementsNullable(this.caseLabels, expressions.stream().map(J.class::cast).collect(toList()))));
+            } else {
+                throw new IllegalStateException("caseLabels contains an entry that is not an Expression, use withCaseLabels instead.");
```

Comment:
This exception will only occur when Java 21+ is involved, we do however need to find a way to convey to recipe authors that they should use `withCaseLabels` instead of  `withExpressions` when their recipe applies to Java 21.